jsx/App/Stories/Story/lib/txt_sync.js

```diff
@@ -11,67 +36,104 @@ export function setupTextSync() {
+            setMediaCurrentTime(newTime);
+        } else {
+            // A file without AV files will not have newTime associated with each sentence.
+            // In this case, use the sentence id as part of the new URL after selecting a sentence. 
+            const sentId = $(this).data('sentence_id');
+            updateForUntimedFile(sentId);
+        }
+        // document.location.search = $(this).data('start_time');
+    });
 
-    for (var i = 0; i < ts_tag_array.length; i++) {
-        ts_start_time_array[i] = ts_tag_array[i].getAttribute("data-start_time");
-        ts_stop_time_array[i] = ts_tag_array[i].getAttribute("data-end_time");
+    /* For files without AV, change the selected sentence's color and scroll to it. */
+    function updateForUntimedFile(sentId) {
+        for (var i = 0; i < ts_tag_array.length; i++) {
+            // sentence id starts with 1
+            if (i+1 == sentId) {
+                ts_tag_array[i].setAttribute("id", "current");
+                scrollIntoViewIfNeeded($("#current")[0]);
+                changeSentenceColorToHighlight(i);
+            } else {
+                changeSentenceColorToNormal(i);
+            }
+        }
+        updateTimestampQuery(sentId);
     }
-    // }
 
     // I/P: t, an integer number of milliseconds
     // O/P: the player updates to the given time
     // Status: untested
-    function jumpToTime(t) {
-        updateTimestampQuery(t);
-        try {
-            const media = $("[data-live='true']").get(0);
+    // This function adjusts the AV file(s)' timestamp according to the 
+    // selected sentence's URL
+    function setMediaCurrentTime(t) {
+        // Change by Holly, October 3, 2020: I moved the updateTimestampQuery call to 
+        // be part of the onclick of .timeStamp.
+        // Also, I removed the try-catch block so that no error is thrown for non-AV files. 
+        const media = $("[data-live='true']").get(0);
+        if (media) {
             media.currentTime = (t + 2) / 1000;
         }
-        catch(err) {
-            console.log(err);
-            console.log("We think you tried to jump to time before the MEDIA element was created.")
+        // try {
+        //     const media = $("[data-live='true']").get(0);
+        //     media.currentTime = (t + 2) / 1000;
+        // }
+        // catch(err) {
+        //     console.log(err);
+        //     console.log("We think you tried to jump to time before the MEDIA element was created.")
+        // }
+    }
+
+    /* Updates the URL according to a sentence's index id. */
+    function updateTimestampQuery(newSentenceIndex) {
+        if (window.history.replaceState) {
+            // const newurl = window.location.protocol + "//" + window.location.host + window.location.pathname + window.location.hash.replace(/\?.*$/, ''); // hacky...
+
+            // (Holly:) The original code had -1 in the timestamp after ? --- why is this -1 necessary? 
```

Comment:
The -1 might have been a hack so that the first sentence doesn't highlight until things actually start playing. That way, the first sentence becomes highlighted 1 ms into the recording if its start time was 0 ms. 

I think the don't-highlight-until-media-starts-playing behavior is still desirable, so please test your code for that. Then go ahead and delete the old code and your comment.